src/hotspot/cpu/aarch64/c2_MacroAssembler_aarch64.cpp

```diff
@@ -1654,3 +1655,17 @@ void C2_MacroAssembler::vector_round_sve(FloatRegister dst, FloatRegister src, F
   sve_fcvtzs(dst, T, ptrue, dst, T);
   // result in dst
 }
+
+bool C2_MacroAssembler::emit_trampoline_stub(int insts_call_instruction_offset, address target) {
+  if (ciEnv::current()->task() != NULL) {
+    assert(is_c2_compile(ciEnv::current()->task()->comp_level()), "not C2 compilation task");
```

Comment:
You don't need this assert. `C2_MacroAssembler` is used only in `.ad` files used by C2 only.

Reply:
Done